packages/node_modules/@ractivejs/cli/src/index.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env node */
+/* eslint-disable no-console */
+import serialize from 'serialize-javascript'
+import program from 'commander'
+import Ractive from '@ractivejs/core'
+import { toParts, toModule } from '@ractivejs/utils-component'
+import { readFromStream, writeToStream, escapeUnicode } from './utils'
+import * as scaffolds from './scaffolds'
+
+program
+  .command('parse')
+  .description('Takes templates and emits its JSON/JS AST form.')
+  .option('-x, --no-csp', 'Do not output expression functions.')
+  .option('-d, --delimiters [pair]', 'Start and end delimiters for normal mustaches.')
+  .option('-s, --static-delimiters [pair]', 'Start and end delimiters for static mustaches.')
+  .option('-t, --triple-delimiters [pair]', 'Start and end delimiters for triple mustaches.')
+  .option('-p, --static-triple-delimiters [pair]', 'Start and end delimiters for static triple mustaches.')
+  .option('-u, --escape-unicode', 'Export non-ASCII characters in strings as UTF escapes.')
+  .action(cmd => {
+    const { csp, delimiters: d, staticDelimiters: s, tripleDelimiters: t, staticTripleDelimiters: p } = cmd
+    const x = d => d ? d.split(' ') : void 0
+    readFromStream(process.stdin).then(template => {
+      const parsed = Ractive.parse(template, { delimiters: x(d), staticDelimiters: x(s), tripleDelimiters: x(t), staticTripleDelimiters: x(p), csp })
+      const serialized = serialize(parsed)
+      const escaped = cmd.escapeUnicode ? escapeUnicode(serialized) : serialized
+      const newlined = `${escaped}\n`
+      return writeToStream(process.stdout, newlined)
+    }).catch(err => {
+      console.error(err.message)
+      process.exit(1)
+    })
+  })
+
+program
+  .command('parts')
+  .description('Takes component files and emits its JSON parts form.')
+  .action(() => {
+    readFromStream(process.stdin).then(component => {
+      return writeToStream(process.stdout, serialize(toParts(component)))
+    }).catch(err => {
+      console.error(err.message)
+      process.exit(1)
+    })
+  })
+
+program
+  .command('module')
```

Comment:
A new mode that I'm finding quite useful with typescript is to take the template, partials, and css and emit all of those that are available as `export const css = $CSS; export const template = $TEMPLATE; export const partials = $PARTIALS;`, though the partials part is a bit dubious, since they're already in the template at that point. It might make sense to have a command that does that too, so that you can have the script portion be separate in whatever compile-to-js you like as long as you support esm somewhere in your build.

Reply:
Hmm... currently, everything is very "SFC-in, CJS-out". Currently it's like:

```js
const module = toModule(transpilePartsWithYourFunction(toParts(componentInJadeScssEsOrTs)))
const constructor = toConstructor(transpilePartsWithYourFunction(toParts(componentInJadeScssEsOrTs)))
```

or in cli

```sh
# component.ractive.html written in Jade, Scss, Ts or whatever
cat component.ractive.html | ractive parse | transpilePartsWithYourCommand | ractive module > component.js
```

If that design works, great. Otherwise, I'll build a command to compile template, styles, script and partials to an ES module (and maybe a CJS variant of this as well). 👍